tests/roles/dataplane_adoption/tasks/main.yaml

```diff
@@ -132,15 +133,99 @@
         rbd_secret_uuid=$CEPH_FSID
     EOF
 
-- name: Create OpenStackDataPlaneNodeSet
+- name: create a custom dataplane service for each additional cell
+  when:
+    - compute_adoption|bool
+    - ('ceph' in [nova_libvirt_backend])
   no_log: "{{ use_no_log }}"
   ansible.builtin.shell: |
     {{ shell_header }}
-    cat <<EOF > edpm-crd.yaml
+    {{ oc_header }}
+    {{ cells_env }}
+
+    for CELL in $(echo $RENAMED_CELLS); do
+      if [ "$CELL" = "cell1" ]; then
+        # Example: NOVA_EDPM_SERVICE="nova-cell1-custom"
+        continue
+      else
+        NOVA_EDPM_SERVICE="nova-$CELL"
+      fi
+      oc apply -f - <<EOF
+    ---
+    apiVersion: dataplane.openstack.org/v1beta1
+    kind: OpenStackDataPlaneService
+    metadata:
+      name: $NOVA_EDPM_SERVICE
+    spec:
+      dataSources:
+        - secretRef:
+            name: ${NOVA_EDPM_SERVICE}-compute-config
+        - secretRef:
+            name: nova-migration-ssh-key
+        - configMapRef:
+            name: nova-ceph-extra-config
+            optional: true
+      playbook: osp.edpm.nova
+      caCerts: combined-ca-bundle
+      edpmServiceType: nova
+      containerImageFields:
+      - NovaComputeImage
+      - EdpmIscsidImage
+    EOF
+    done
+
+- name: Create the OpenStackDataPlaneNodeSet CRs for each Nova compute cell
+  no_log: "{{ use_no_log }}"
+  ansible.builtin.shell: |
+    {{ shell_header }}
+    {{ nodesets_env }}
+    {{ edpm_computes_shell_vars_src }}
+
+    for CELL in $(echo $RENAMED_CELLS); do
+      if [ "$CELL" = "cell1" ]; then
+        # Example: NOVA_EDPM_SERVICE="nova-cell1-custom"
+        NOVA_EDPM_SERVICE="nova"
+        OSDPNS="openstack"
+      else
+        NOVA_EDPM_SERVICE="nova-$CELL"
+        OSDPNS="openstack-$CELL"
+      fi
+
+      computes=COMPUTES_$(echo ${CELL}|tr '[:lower:]' '[:upper:]')[@]
+      for compute in $computes; do
```

Comment:
Am not sure about this magic, or simply document for users to manually adjust the nodes section, like I did in some other places for cells.